### PR TITLE
Update user_pseudo_id to client_key

### PR DESCRIPTION
**Pull request by TannerHopkins**
## Description & motivation
There was a [recent issue raised](https://github.com/Velir/dbt-ga4/issues/163) for `user_pseudo_id` failing unique checks for multi-property implementations of the package. A decision was made to replace `user_pseudo_id` with `client_key` throughout the package, with `client_key` being a hash of `user_pseudo_id` and `stream_id`.
This PR accomplishes the planned work laid out in that issue, substituting `client_key` for `user_pseudo_id` throughout the package. 

A few notes:
- `user_pseudo_id` was only kept in the base_select.sql macro
- Whenever both `user_pseudo_id`  and `stream_id` were used in a group by statement, I substituted out _both fields_ for `client_key` to avoid redundancy
- I had to rename a few models that included user_pseudo_id in the name. I'm assuming this would cause breaking changes for some users of this package, so I'm curious if the new release that would include this PR would need to be a new major version (i.e. 4.0.0)?

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have run `dbt test` and `python -m pytest .` to validate existing tests

### CODE REVIEW

**Comment by TannerHopkins** [comment]
Just pushed one more commit to include this description for client_key whenever it is mentioned in YML files throughout the package:
```description: Hashed combination of user_pseudo_id and stream_id```
Feel free to suggest a better description, or let me know if we don't need this description everywhere. I at least wanted to replace the old description for `user_pseudo_id` from one of the YML files, and thought it might be a good idea to just add this everywhere while I'm at it since this new field isn't native to ga4.

File: models/marts/core/fct_ga4__sessions_daily.sql
Comment (dgitis) [inline models/marts/core/fct_ga4__sessions_daily.sql:1]: Revert `stream_id` and `group by` and re-add `stream_id` to the `join_metrics_and_conversions` cte.

File: models/marts/core/fct_ga4__user_ids.sql
Comment (dgitis) [inline models/marts/core/fct_ga4__user_ids.sql:1]: Revert `stream_id` and `group by`.

**Comment by TannerHopkins** [comment]
> Thanks for all of your hard work here. I was literally sitting down to resolve this myself. The PR looks good except for one issue affecting several files.
> 
> The `stream_id` is needed for multi-site installations and GA4 properties that combine app and web data to segment streams when reporting.
> 
> Fix that issue and then I'd like to run this on some test data before approving just because it touches a lot files.

Thanks for calling that out! I had wondered if removing that field would be an issue or not, was hoping it would be mentioned in PR review if so. For context, I haven't used this package a whole lot just yet - trying to get it set up for a very large multi-property implementation though (hundreds of properties) and @adamribaudo  mentioned this issue might be a blocker.

I'll fix this now!

**Review by dgitis — APPROVED** [review]
That looks good @TannerHopkins. Thanks for helping out.

File: macros/base_select.sql [hunk 1]
```diff
@@ -42,6 +43,7 @@
     event_server_timestamp_offset,
     user_id,
     user_pseudo_id,
+    to_base64(md5(concat(user_pseudo_id, stream_id))) as client_key,
```
Comment (adamribaudo-velir) [inline macros/base_select.sql:46]: @TannerHopkins we do all other key creation in `stg_ga4__events`. Could we move that there to a CTE early in the model?

**Pull request merged**